### ISSUE TITLE
Still bug with handling decimal point

### ISSUE THREAD

**Issue by rjo12** [Darkempire78/OpenCalc#133]
I reported a similar bug recently, which seems to have been closed.

This bug is in OpenCalc v2.0.0-beta3 which (as I write this) is the latest version available through F-Droid (added 2 December 2022).

It is easily triggered by entering any expression with more than one numeric value, one of which has a decimal point without a leading zero, and one which has leading digits followed by a decimal point.

For example,  the expression (typed)  as 1.x.1 is redisplayed as 10.x0.1 and gives a result of 1 (not 0.1 which would be correct).

1.9x.2 (typed) is redisplayed as 10.9x0.2 which gives a result of 2.18 (not 0.38, which would be correct).

General comment - the behaviour I'm seeing appears to be due to code trying to handle input when a value with a decimal point is entered without any leading digits.   The code reading/scanning the input seems to replace all instances of decimal point (".") in the input with the text "0."    And it does this as the user hits keys.    That is inappropriate since, as the user hits more keys, the user input gets "adjusted" repeatedly and, eventually, the modified user inpout is mathematically different to the expression entered by the user.

Furthermore, the behaviour is visible with partial expressions.  Entering 1.x. is displayed as 10.x0. (note the additional zeros) and typing the next digit highlights the problem (producing incorrect results).   As noted above, entering 1.x.1 is displayed as 10.x0.1 which gives an incorrect result.

This observed behaviour suggests the code [which I have not examined] is organised with two parts - Part A (my label, for sake of discussion) replaces all occurrences of "." with "0."  and Part B tries (unsuccessfully) to correct resultant problems.    Presumably Part B was written to handle some of the detected/reported bugs like I am reporting here.    The two-step approach of "add of zeros on decimal" (Part A) followed by "try to fix the mess" (Part B) will never totally succeed (there will always be cases to trip the user up, by giving a wrong result, and regular new bug reports associated with those cases [or users giving up on the app, since this is a rather glaring flaw in a calculator app]).    The "Part A" code needs to be corrected, and the "Part B" can be simplified (possibly removed entirely).

**Comment by Darkempire78** [comment]
@rjo12 The issue seems to be fixed in the v2.0.0-beta4, can you verify please ?
https://github.com/Darkempire78/OpenCalc/releases/tag/v2.0.0-beta4

**Comment by rjo12** [comment]
Apologies for delay in reply - I've been offline for the last couple of days.

The symptom described above is still present in v2.0.0-beta4  (the two examples I described previously still have the same erroneous effect)

However, I did notice there is a pre-release version of v2.0.0-beta5, so downloaded and installed that version.    It gives correct results for the same two examples.

I'll keep testing to see if any other problems become evident.   But happy to concur with you marking this issue as closed based on behaviour of beta5.   If I do find further quirks, I'll start a new issue.
